blocks/editable/index.js

```diff
@@ -72,6 +75,28 @@ function getFormatProperties( formatName, parents ) {
 
 const DEFAULT_FORMATS = [ 'bold', 'italic', 'strikethrough', 'link' ];
 
+/**
+ * Transforms internal block's representation into an Element.
+ *
+ * @param {Array} value Value to transform
+ * @return {WPElement} Element.
+ */
+export function valueToElement( value ) {
+	if ( ! Array.isArray( value ) ) {
+		return value;
+	}
+
+	return value.map( ( element, i ) => {
+		if ( typeof element === 'string' ) {
+			return element;
+		}
+
+		const [ type, props, ...children ] = element;
+
+		return createElement( type, { ...props, key: i }, ...valueToElement( children ) );
```

Comment:
I assume `key` here might not be unique when recursing into children. Not sure how smart React is about differentiating `key` based on child structure, or if it must be unique for the component.

Reply:
I think we should try to get rid of this code, see https://github.com/WordPress/gutenberg/pull/3048#issuecomment-340751231. Or will we be okay with the warnings?

Reply:
> Not sure how smart React is about differentiating `key` based on child structure, or if it must be unique for the component.

It needs to be unique only for the component. See https://reactjs.org/docs/lists-and-keys#keys-must-only-be-unique-among-siblings:

> Keys used within arrays should be unique among their siblings. However they don’t need to be globally unique. We can use the same keys when we produce two different arrays